policy/policy-authority.go

```diff
@@ -145,7 +145,6 @@ func (pa PolicyAuthorityImpl) WillingToIssue(id core.AcmeIdentifier, regID int64
 		}
 	}
 
-	domain = strings.ToLower(domain)
```

Comment:
There's a couple of XXX on line 131-135 that can be removed now, and probably replaced with "Input must be lower-case."